pkg/summarizer/naiveanalyzer/analyzer.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package naiveanalyzer
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+
+	summarypb "github.com/GoogleCloudPlatform/testgrid/pb/summary"
+	"github.com/GoogleCloudPlatform/testgrid/pkg/summarizer/common"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/sirupsen/logrus"
+	"vbom.ml/util/sortorder"
+)
+
+const analyzerName = "naiveanalyzer"
+
+var (
+	jailedRegex = regexp.MustCompile(`(JAILED )?(\/?\/?[A-Za-z0-9\/_\-.:}{]+) - \[([A-Za-z0-9\/:\-_.\s]+)\]\s*`)
+)
+
+// IntString is for sorting, primarily intended for map[string]int as implemented below
+type IntString struct {
+	s string
+	i int
+}
+
+// NaiveAnalyzer implements functions that calculate flakiness as a ratio of failed tests to total tests
+type NaiveAnalyzer struct {
+}
+
+// GetFlakiness returns a HealthinessInfo message with data to display flakiness as a ratio of failed tests
+// to total tests
+func (na *NaiveAnalyzer) GetFlakiness(gridMetrics []*common.GridMetrics, minRuns int, startDate int, endDate int, tab string) *summarypb.HealthinessInfo {
+	testByEnv := make(map[string]*summarypb.TestInfo)
+	infraIssues := make(map[string]int64)
+	for _, test := range gridMetrics {
+		name, env := getNameAndEnvFromTest(test.Name, tab)
+		if len(test.InfraFailures) > 0 {
+			for errorType, errorCount := range test.InfraFailures {
+				infraIssues[test.Name+"-"+errorType] += int64(errorCount)
+			}
+		}
+
+		testInfo, success := calculateNaiveFlakiness(test, minRuns)
+		if !success {
+			continue
+		}
+		testInfo.Name = name
+		testInfo.Env = env
```

Comment:
So, let's don't worry about env for now, set `testInfo.Name = test.Name` here, and we can drop the testByEnv map for now. Maybe drop a TODO for follow ups.

Reply:
Okay, I'll add each TestInfo into an array instead because the TestInfo's still end up as a field in HealthinessInfo. Should I also remove "Env" as a field from TestInfo or just leave it as is to use later?

Reply:
yeah, we can reserve the env field for later.

Reply:
Following up from offline discussion; I think we're fine removing fields from the proto to be clear, and adding them back in with later implementations. (The proto fields can be out of order and logically grouped together, regardless of number, so we don't have to reserve them ahead of time).

Reply:
Done